core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java

```diff
@@ -0,0 +1,66 @@
+package com.baeldung.iostreams;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class DataStreamUnitTest {
+
+    private static final String dataProcessingTextFile = "src/test/resources/iostreams/TestFile.txt";
+    private static final String dataProcessingImageFile = "src/test/resources/iostreams/image.png";
+    private static final String dataProcessingByteStreamFile = "src/test/resources/iostreams/ps.png";
+    private static final String dataProcessingCharStreamFile = "src/test/resources/iostreams/pw.png";
+
+    private static final String textFileContent = "Hello, world!";
+
+    @Test
+    public void giveFile_writeUsingByteStream() throws IOException {
+        DataStream dataStream = new DataStream();
+        dataStream.textDataProcessingByteStream(dataProcessingTextFile, textFileContent);
+
+        File file = new File(dataProcessingTextFile);
+        assertTrue(file.exists());
+        assertEquals(FileUtils.readFileToString(file, "utf-8"), textFileContent);
+
+        Files.delete(Paths.get(dataProcessingTextFile));
+    }
+
+    @Test
+    public void giveFile_writeUsingCharStream() throws IOException {
+        DataStream dataStream = new DataStream();
+        dataStream.textDataProcessingCharStream(dataProcessingTextFile, textFileContent);
+
+        File file = new File(dataProcessingTextFile);
+        assertTrue(file.exists());
+        assertEquals(FileUtils.readFileToString(file, "utf-8"), textFileContent);
+
+        Files.delete(Paths.get(dataProcessingTextFile));
+    }
+
+    @Test
+    public void giveFiles_writeNonTextDataWithStreams() throws IOException {
```

Comment:
The test method name must be: givenXXX_whenYYYY_thenZZZZ being the when and then mandatory and the given optional.

Reply:
Fixed unit test names. Thanks for pointing that out.